src/main/java/cz/cvut/kbss/termit/service/term/OrphanedInverseTermRelationshipRemover.java

```diff
@@ -0,0 +1,75 @@
+package cz.cvut.kbss.termit.service.term;
+
+import cz.cvut.kbss.termit.dto.TermInfo;
+import cz.cvut.kbss.termit.model.Term;
+import cz.cvut.kbss.termit.persistence.dao.TermDao;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Component
+public class OrphanedInverseTermRelationshipRemover {
```

Comment:
The same logic seems to be applicable for exactMatches (and later e.g. for closeMatches) - parameterizing the logic might be valuable.

Reply:
Wasn't part of this task but ok, implemented.